README.md
# APHorizontalMenu

[![Version](http://cocoapod-badges.herokuapp.com/v/APHorizontalMenu/badge.png)](http://cocoadocs.org/docsets/APHorizontalMenu)
[![Platform](http://cocoapod-badges.herokuapp.com/p/APHorizontalMenu/badge.png)](http://cocoadocs.org/docsets/APHorizontalMenu)

## Usage

To run the example project; clone the repo, and run `pod install` from the Example directory first.

## Installation

APHorizontalMenu is available through [CocoaPods](http://cocoapods.org), to install
it simply add the following line to your Podfile:

    pod "APHorizontalMenu"

## Example

![iPhone](https://raw.githubusercontent.com/apascual/APHorizontalMenu/master/Images/APHorizontalMenu.gif)

## Usage

Using APHorizontalMenu is really easy, there is an Example project that you can check, but here are the basics.

You can add APHorizontalMenu programmatically or using Storyboards.

### Programmatically

Just init the APHorizontalMenu, fill it with values and add it to an existing view.

```objc
APHorizontalMenu *horizontalMenu = [[APHorizontalMenu alloc] initWithFrame:CGRectMake(0, 200, 320, 40)];
horizontalMenu.delegate = self;
horizontalMenu.values = @[@"Item 1", @"Item 2", @"Item 3", @"Item 4", @"Item 5", @"Item 6", @"Item 7", @"Item 8", @"Item 9", @"Item 10"];
[self.view addSubview:horizontalMenu];
```

### Storyboards

In this case, create a new UIView using the Storyboards UI designer and set the position, size and layout restrictions as desired. Then, go to the Utilities -> Identity inspector -> Custom class -> Class and write down "APHorizontalMenu".

Then, create an outlet as follows in your Controller.

```objc
@property (weak, nonatomic) IBOutlet APHorizontalMenu *horizontalMenu;
````

And in the implementation file of your controller add at least this.

```objc
self.horizontalMenu.delegate = self;
````

### Delegate

Do not forget to add implement the APHorizontalMenuDelegate as follows. First add the Delegate in the header file of your controller, for example:

```objc
@interface ViewController : UIViewController <APHorizontalMenuSelectDelegate>
````

And conform the protocol by creating the method in the implementation file of your controller so you can receive messages:

```objc
- (void)horizontalMenu:(id)horizontalMenu didSelectPosition:(NSInteger)index {
    NSLog(@"APHorizontalMenu selection: %d", index);
    // Do whatever
}
````

### Customization

You can customize some of the properties of APHorizontalMenu like this:

```objc
self.horizontalMenu.cellBackgroundColor = [UIColor brownColor];
self.horizontalMenu.cellSelectedColor = [UIColor greenColor];
self.horizontalMenu.textColor = [UIColor blackColor];
self.horizontalMenu.textSelectedColor = [UIColor blueColor];
self.horizontalMenu.selectedIndex = 2;
self.horizontalMenu.visibleItems = 3;
```

## Contact

Abel Pascual

* Mail: [abelpascual@gmail.com](mailto:abelpascual@gmail.com)
* Twitter: [@Abel_Pascual](https://twitter.com/Abel_Pascual)

## License

APHorizontalMenu is available under the MIT license. See the LICENSE file for more info.



CHANGELOG.md
# APHorizontalMenu CHANGELOG

## 0.1.0

Initial release.


Example/APHorizontalMenu/AppDelegate.m
//
//  AppDelegate.m
//  APHorizontalMenu
//
//  Created by Abel Pascual on 01/05/14.
//  Copyright (c) 2014 Abel Pascual. All rights reserved.
//

#import "AppDelegate.h"

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
    // Override point for customization after application launch.
    return YES;
}
							
- (void)applicationWillResignActive:(UIApplication *)application
{
    // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
    // Use this method to pause ongoing tasks, disable timers, and throttle down OpenGL ES frame rates. Games should use this method to pause the game.
}

- (void)applicationDidEnterBackground:(UIApplication *)application
{
    // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later. 
    // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
}

- (void)applicationWillEnterForeground:(UIApplication *)application
{
    // Called as part of the transition from the background to the inactive state; here you can undo many of the changes made on entering the background.
}

- (void)applicationDidBecomeActive:(UIApplication *)application
{
    // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
}

- (void)applicationWillTerminate:(UIApplication *)application
{
    // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
}

@end


Example/APHorizontalMenu/ViewController.m
//
//  ViewController.m
//  APHorizontalMenu
//
//  Created by Abel Pascual on 01/05/14.
//  Copyright (c) 2014 Abel Pascual. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIImage *image = [UIImage imageNamed:@"large1"];
    UIImageView *view = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
    view.image = image;
    self.values = @[@"Item 1", image, @"Item 3", view, @"Item 5", @"Item 6", @"Item 7", @"Item 8", @"Item 9", @"Item 10"];
    
	self.horizontalMenu.values = self.values;
    self.horizontalMenu.delegate = self;
    
    // Optional settings
    //self.horizontalMenu.cellBackgroundColor = [UIColor brownColor];
    //self.horizontalMenu.cellSelectedColor = [UIColor greenColor];
    //self.horizontalMenu.textColor = [UIColor blackColor];
    //self.horizontalMenu.textSelectedColor = [UIColor blueColor];
    //self.horizontalMenu.selectedIndex = 2;
    //self.horizontalMenu.visibleItems = 3;
    
    APHorizontalMenu *menu2 = [[APHorizontalMenu alloc] initWithFrame:CGRectMake(0, 200, 320, 40)];
    menu2.values = self.values;
    [self.view addSubview:menu2];
}

- (void)horizontalMenu:(id)horizontalMenu didSelectPosition:(NSInteger)index {
    NSLog(@"APHorizontalMenu selection: %ld", (long)index);
}

@end
